lib/Conversion/TritonGPUToLLVM/ViewOpToLLVM.cpp

```diff
@@ -316,6 +316,30 @@ struct MemDescTransOpConversion
   }
 };
 
+struct MemDescReshapeOpConversion
+    : public ConvertOpToLLVMPattern<MemDescReshapeOp> {
+  using ConvertOpToLLVMPattern::ConvertOpToLLVMPattern;
+  LogicalResult
+  matchAndRewrite(MemDescReshapeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op->getLoc();
+    auto resultTy = cast<TensorOrMemDesc>(op.getType());
+    auto llvmElemTy =
+        getTypeConverter()->convertType(resultTy.getElementType());
+    auto srcSmemObj = getSharedMemoryObjectFromStruct(loc, adaptor.getSrc(),
+                                                      llvmElemTy, rewriter);
+    SmallVector<Value> offsets = srcSmemObj.getOffsets();
+    auto b = TritonLLVMOpBuilder(loc, rewriter);
+    // TODO: handle offsets correctly.
```

Comment:
What is not handled correctly?

Reply:
oops I forgot to add that. I was missing the linearization/delinearization to update the offsets